zipkin/src/zipkin_reporter_impl.h

```diff
@@ -49,9 +49,46 @@ class ReporterImpl : public Reporter {
 
   bool flushWithTimeout(std::chrono::system_clock::duration timeout) override;
 
+  uint64_t droppedSpanCount() const override;
+
+  uint64_t getAndResetDroppedSpanCount() override;
+
+  /**
+   * @brief getReportPeriod Returns the period of time to wait between auto
+   * flushes
+   * @return Setting for the duration between auto flushes
+   */
+  SteadyClock::duration reportPeriod() const override {
+    return reporting_period_;
+  }
+
+  /**
+   * @brief setReportPeriod Adjust the period of time to wait between auto
+   * flushes
+   * @param report_period The new size of the span buffer
+   */
+  void setReportPeriod(SteadyClock::duration reporting_period) override {
+    reporting_period_ = reporting_period;
```

Comment:
This has a race condition, no? Since `reporting_period_` is read from another thread.